### PR TITLE
feat(lint): add PsyC+JSON lint and autocomplete

### PR DESCRIPTION
**Pull request by superwhoopy**
Configure ALE and coc.nvim for linting of PsyC files using *psyko*, and
linting and autocompletion of JSON configuration files, using a JSON
language server.

In other news:

- add documentation (keyword `vim-asterios`)
- add a `tests/` directory, with only "manual" tests at the moment
- add an editorconfig file
- update and cleanup README file

### CODE REVIEW

File: doc/vim-asterios.txt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ASTERIOS application. It includes out of the box syntax highlighting for:
+
+  * the PsyC (support of the `.psy` files)
+  * the Bgt (support of the `.bgt` files)
+  * the KhiC (support of the `.khic` files) - deprecated
+  * the tak (support of the `.tak` files)   - deprecated
+
+For ASTERIOS K19.2 and higher, vim-asterios also features an integration with
+`ALE` (Asynchronous Lint Engine) to provide linting of PsyC source files, and
+an integration with `coc.nvim` for linting and autocompletion of JSON
```
Comment (jeanguyomarch) [inline doc/vim-asterios.txt:61]: That's really a burden to install nodejs :/

Comment (superwhoopy) [reply doc/vim-asterios.txt:61]: Agreed. Unfortunately, there is currently no way around it: the only serious language server for JSON available is the one from Visual Studio Code...

File: doc/vim-asterios.txt [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+:AsteriosDiag                                                  *:AsteriosDiag*
+
+  Print a diagnostics message of the status of the ASTERIOS plug-in.
+
+
+==============================================================================
+ACKNOWLEDGMENTS                                *vim-asterios-acknowledgements*
+
+All Vim and Neovim lovers at Krono-Safe would like to thank the maintainers
+and contributors of the following projects - because life would just be less
```
Comment (jeanguyomarch) [inline doc/vim-asterios.txt:213]: Less fun, indeed.

**Review by jeanguyomarch — APPROVED** [review]
I find painful trying to configure all these, but the result totally worths it. Great job. :100:

**Comment by superwhoopy** [comment]
Thanks alot @jeanguyomarch !

**Pull request merged**